aya/src/maps/mod.rs

```diff
@@ -174,6 +177,16 @@ pub enum MapError {
         io_error: io::Error,
     },
 
+    /// GetMapFromFD failed
+    #[error("`{call}` failed when getting map's fd")]
+    GetMapFDError {
```

Comment:
I don't think we should add this variant, and use SyscallError instead?

Reply:
Soudns good I only did this because in this case (with `BPF_OBJ_GET_INFO_BY_FD`) we don't seem to get an error code

Reply:
I think we can remove the `code` field, it was there before I added `io_error` but then `io_error` made it redundant

Reply:
Done